src/Amazon.CloudWatch.EMF/Model/MetricsContext.cs

```diff
@@ -97,29 +103,37 @@ public bool HasDefaultDimensions
         /// Multiple calls using the same key will be stored as an array of scalar values.
         /// </summary>
         /// <example>
-        /// metricContext.PutMetric("Latency", 100, Unit.MILLISECONDS)
+        /// Defaults to StorageResolution : metricContext.PutMetric("Latency", 100)
+        /// Standard Resolution metric : metricContext.PutMetric("Latency", 100, Unit.MILLISECONDS)
+        /// High Resolution metric : metricContext.PutMetric("Latency", 100, Unit.MILLISECONDS,StorageResolution.HIGH)
         /// </example>
         /// <param name="key">the name of the metric</param>
         /// <param name="value">the value of the metric</param>
         /// <param name="unit">the units of the metric</param>
-        public void PutMetric(string key, double value, Unit unit)
+        /// <param name="storageResolution">the storage resolution of the metric. Default Set to StandardResolution</param>
+        public void PutMetric(string key, double value, Unit unit, StorageResolution storageResolution = StorageResolution.STANDARD)
         {
-            Validator.ValidateMetric(key, value);
-            _metricDirective.PutMetric(key, value, unit);
+            Validator.ValidateMetric(key, value, storageResolution, _storageResolutionMetrics);
+            _metricDirective.PutMetric(key, value, unit, storageResolution);
+            if (!_storageResolutionMetrics.ContainsKey(key))
```

Comment:
Can we move these two lines that validate metrics in ValidateMetric() method? It will be good to have all validations in one place.